src/coord/src/catalog.rs

```diff
@@ -2713,6 +2771,30 @@ impl<S: Append> Catalog<S> {
                     });
                     actions
                 }
+                Op::UpdateComputeInstanceStatus {
+                    compute_id,
+                    replica_id,
+                    process_id,
+                    status,
+                } => {
+                    let status_known = self
+                        .state
+                        .try_get_compute_instance_status(compute_id, replica_id, process_id)
+                        .is_some();
+                    if status_known {
+                        let update = self.state.pack_compute_instance_status_update(
+                            compute_id, replica_id, process_id, -1,
+                        );
+                        builtin_table_updates.push(update);
+                    }
+
+                    vec![Action::UpdateComputeInstanceStatus {
```

Comment:
this could get moved up into `if status_known` and return an empty vec otherwise; I slightly prefer that because it holds a slightly tighter invariant on understanding if something falls out of sync.

Reply:
But that would mean the `mz_cluster_replicas_status` table always stays empty! When a replica is first created, it has no entries in that table. So when the orchestrator reports a status for the first time and we handle the `Op::UpdateComputeInstanceStatus` here, `status_known` will be `false`, but we still want to put the new status into the table, so the `Action::UpdateComputeInstanceStatus` is necessary.

Reply:
Ah, ok--I clearly didn't follow, so think comments will help folks in the future.

Reply:
Good point, I'll add one here too!